src/Symfony/Component/Uid/Uuid.php

```diff
@@ -130,8 +121,19 @@ final public static function v8(string $uuid): UuidV8
         return new UuidV8($uuid);
     }
 
-    public static function isValid(string $uuid): bool
+    /**
+     * @param int-mask-of<Uuid::FORMAT_*> $format
+     */
+    public static function isValid(string $uuid /*, int $format = self::FORMAT_RFC_4122 */): bool
     {
+        $format = 1 < \func_num_args() ? func_get_arg(1) : self::FORMAT_RFC_4122;
+
+        if (36 === \strlen($uuid) && !($format & self::FORMAT_RFC_4122)) {
+            return false;
+        }
+
+        $uuid = self::transformToRfc4122($uuid, $format);
+
```

Comment:
Hi @alexandre-daubois  sorry to dig this up, but while 7.2 is not release.. 

I find this early condition non "optimal", as default case will be in fact $format = FORMAT_RFC_4122.. 

What if we did something like this here (before? instead?),  avoiding the regexp if the string is not 36 chars long ? 

```php
    if (($format & self::FORMAT_RFC_4122) && 36 !== \strlen($uuid)) {
        return false;
    }
```

What do you think ?

Reply:
(before line 131, i messed up my selection 😅 )

Reply:
I did try a few things but it would require some refactoring because of how the control flow is currently. I'm not sure it would worth it and if performances would greatly improve :thinking:

Reply:
Yep that is very true!